### PR TITLE
Task 12 create institute model

### PR DESCRIPTION
**Pull request by josh-cooper**
## Description
Create institute model with the following attributes: name.

## Breakdown of Commits
Create institute model : 2db03dfbe05dc64739de5ebe26e4a8581017cd70

Migrate institute model : 7b8cb1ff0f2ac4279ffad9f7ac57e5afd8bc95c6

Add institute factories : 86e773362c557fb551b4d49cb6b4cb27b955d769

Add institute specs : 0a07ff2e345ae3b38aabf381bb968b2f71f272a7

#### TODO Checklist
* [x] Create institute model with name attr

### CODE REVIEW

File: my-sms/app/controllers/courses_controller.rb [hunk 1]
```diff
@@ -94,4 +94,8 @@ def load_course
   def flash_errors
     super(@course)
   end
+
+  def flash_errors
+    super(@course)
+  end
```
Comment (jhuizy) [inline my-sms/app/controllers/courses_controller.rb]: repeated method?

Comment (josh-cooper) [reply my-sms/app/controllers/courses_controller.rb]: Don't know how that snuck in there. Removed that commit.

File: my-sms/spec/factories/institutes.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+require 'faker'
+
+FactoryBot.define do
+  factory :institute do
+    sequence(:name) { |n| "Institute #{n}" }
+  end
+  factory :institute_demo, class: :institute do
```
Comment (jhuizy) [inline my-sms/spec/factories/institutes.rb]: specify as a trait maybe, or just update the name logic for the main/parent factory

File: my-sms/spec/models/institute_spec.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
+
+RSpec.describe Institute, :type => :model do
+  subject { institute }
+  let(:institute) { build(:institute, institute_params) }
+  let(:institute_params) { {} }
+
+  context 'with valid params' do
```
Comment (jhuizy) [inline my-sms/spec/models/institute_spec.rb:10]: test validity of the factory

```ruby
{ is_expected.to be_valid }

Comment (josh-cooper) [reply my-sms/spec/models/institute_spec.rb:10]: Fixed in cb06b09f0dd87be98de00cf28dce789da850fe1b

File: my-sms/spec/factories/institutes.rb (hunk 2 above)
Comment (josh-cooper) [reply my-sms/spec/factories/institutes.rb]: Refactored it into a trait in 52bc9b0b2a26ce81550b50e67027df0bb279029b
I like having the more deterministic factory for testing.

**Review by jhuizy — APPROVED** [review]

**Pull request merged**